force-app/main/default/classes/ProgramSelector_TEST.cls

```diff
@@ -40,6 +42,68 @@ public with sharing class ProgramSelector_TEST {
         }
     }
 
+    @IsTest
+    private static void testGetActiveProgramCohortsFromProgramId() {
+        String methodName = 'getBuckets';
+        List<String> buckets = new List<String>{ ACTIVE_BUCKET };
+        Schema.SObjectType programCohortSObjType = ProgramCohort__c.SObjectType;
+        Schema.SObjectField statusField = ProgramCohort__c.Status__c;
+
+        TestDataFactory.generateProgramData();
+        Id programId = TestDataFactory.programs[0].Id;
+
+        List<Bucket__mdt> cohortStatusBuckets = createBuckets();
+
+        //Insert a cohort with different status so we check that we return
+        //program cohorts in active status
+        ProgramCohort__c cohort = new ProgramCohort__c();
+        cohort.Name = 'New Cohort';
+        cohort.Program__c = programId;
+        cohort.StartDate__c = System.today();
+        cohort.Status__c = 'Planned';
+
+        insert cohort;
+
+        List<ProgramCohort__c> actualCohorts = new List<ProgramCohort__c>();
+
+        bucketSelectorStub = new StubBuilder(FieldBucketSelector.class)
+            .when(
+                methodName,
+                List<String>.class,
+                Schema.SObjectType.class,
+                Schema.SObjectField.class
+            )
+            .calledWith(buckets, programCohortSObjType, statusField)
+            .thenReturn(cohortStatusBuckets)
+            .build();
+
+        programSelector.bucketSelector = (FieldBucketSelector) bucketSelectorStub.create();
+
+        Test.startTest();
+        actualCohorts = programSelector.getProgramCohortsFromProgramId(programId);
+        Test.stopTest();
+
+        System.assertEquals(
+            1,
+            actualCohorts.size(),
+            'Actual results not the same as expected results'
```

Comment:
Can we be more specific in these messages for example Expected one active cohort to be returned? Also if we know there is only one do we need to loop it or just analyze that one record?

Reply:
I was debating it and then decided to add the loop. I will remove the loop